packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics.util.ts

```diff
@@ -406,20 +412,28 @@ describe('internal-plugin-metrics', () => {
       });
     });
 
-    it('calls getBuildType correctly', () => {
-      const getBuildTypeSpy = sinon.spy(CallDiagnosticUtils, 'getBuildType');
-      const markAsTestEvent = true;
-      const webClientDomain = 'https://web.webex.com';
-
-      // just submit any event
-      prepareDiagnosticMetricItem(webex, {
+    it('getBuildType returns correct value', () => {
+      const item: any = {
         eventPayload: {
-          event: {name: 'client.exit.app', eventData: {markAsTestEvent, webClientDomain}},
+          event: {
+            name: 'client.exit.app',
+            eventData: {
+              markAsTestEvent: true,
+              webClientDomain: 'https://web.webex.com'
+            }
+          },
         },
         type: ['diagnostic-event'],
-      });
+      };
+
+      // just submit any event
+      prepareDiagnosticMetricItem(webex, item);
+      assert.deepEqual(item.eventPayload.origin.buildType, 'test');
 
-      assert.calledOnceWithExactly(getBuildTypeSpy, webClientDomain, markAsTestEvent);
+      delete item.eventPayload.origin.buildType;
+      item.eventPayload.event.eventData.markAsTestEvent = false;
```

Comment:
Since it is listed in `getBuildType` only, I also didn't test all the cases here. 

We're just ensuring getBuildType is called with appropriate arguments and that's why wanted to alter the argument and see if we get the other outcome.

Reply:
yeah, but as we are unable to really verify that `prepareDiagnosticMetricItem()` calls `getBuildType()`, then in theory we shouldn't be relying on the fact that getBuildType tests pass to imply that `prepareDiagnosticMetricItem()` also works. In theory `prepareDiagnosticMetricItem()` might be executing any other code that happens to return the right result for the 2 values of `markAsTestEvent` but not for the rest of webClientDomain or NODE_ENV values. Like I said, it's just theoretical, in practice it's very unlikely that `prepareDiagnosticMetricItem()` would do something else instead of calling `getBuildType()` and still work correctly just for `markAsTestEvent`, that's why I've approved the PR.

Reply:
Makes sense. I hope we do not land in such a situation. 

Thanks @marcin-bazyl